### ISSUE TITLE
[BUG] dropout_p may be necessary for scaled_dot_product_attention

### ISSUE THREAD

**Issue by Dicer-Zz** [THUDM/ChatGLM2-6B#294]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I noticed that ChatGLM2 used FlashAttention by utilizing `torch.nn.functional.scaled_dot_product_attention` at [modeling_chatglm.py#L224](https://huggingface.co/THUDM/chatglm2-6b/blob/162b620e3078b03eefff94eb5f762d4093425fb5/modeling_chatglm.py#L224):

```python
context_layer = torch.nn.functional.scaled_dot_product_attention(query_layer, key_layer, value_layer, is_causal=True)
```

However, I think that passing argument `dropout_p` for `attention_dropout` seems to be missing here.

Also, I checked that the current value of `attention_dropout` is 0.0 in `
configuration_chatglm.py`, which is the same as the default value of `dropout_p`, so this bug has not had any actual impact yet.

### Expected Behavior

Passing `attention_dropout` to `scaled_dot_product_attention` may be necessary.

### Steps To Reproduce

See current behavior section.

### Environment

```markdown
- ChatGLM2: v1.0
```

**Comment by lilongxian** [comment]
可以要也可以不要，参考了vicuna   longchat ,这些也都没要

**Comment by libreliu** [comment]
> 我也刚好看到了这块的 code...发现楼主提过了

出于和下面行为
```
attention_probs = self.attention_dropout(attention_probs)
```
对齐的考虑，感觉这里应该是要 dropout 的，否则微调时其实没有 dropout 参与了

另一个注意到的问题是，`torch.nn.functional.scaled_dot_product_attention` 目前是不认 `module.training` 的，需要手动把 dropout_p 在 eval 时旁路掉
